pages/tutorials/yaml/phases.rst

```diff
@@ -0,0 +1,294 @@
+.. slug: phases
+.. title: Phases and their Interfaces
+.. highlight:: yaml
+
+.. jumbotron::
+
+   .. raw:: html
+
+      <h1 class="display-3">Phases and their Interfaces</h1>
+
+   .. class:: lead
+
+      A description of how phases and interfaces are defined in YAML input files
+
+Phases
+======
+
+For each phase that appears in a problem, a corresponding entry should be
+present in the input file(s).
+
+Phase Attributes
+----------------
+
+Phase Name
+^^^^^^^^^^
+
+The ``name`` entry is a string that identifies the phase. It must be unique
+within the file among all phase definitions of any type. Phases are referenced
+by name when importing them.
+
+
+The Thermodynamic Model
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The thermodynamic model used to represent a phase is specified in the ``thermo``
+field. Supported models are:
+
+    - `binary-solution-tabulated <https://cantera.org/documentation/dev/sphinx/html/yaml/phases.html#binary-solution-tabulated>`__
+    - `compound-lattice <https://cantera.org/documentation/dev/sphinx/html/yaml/phases.html#compound-lattice>`__
+    - `constant-density <https://cantera.org/documentation/dev/sphinx/html/yaml/phases.html#constant-density>`__
+    - `Debye-Huckel <https://cantera.org/documentation/dev/sphinx/html/yaml/phases.html#debye-huckel>`__
+    - `edge <https://cantera.org/documentation/dev/sphinx/html/yaml/phases.html#edge>`__
```

Comment:
Do we have a better way of linking into the dev docs? Using ``` :ref:`section-name` ``` only links to the stable docs. Or even, how does one add a new magic command like ``` :yaml:`section-name` ```?

Reply:
To add a new magic command like `:yaml:`, you need to edit the `parse_docs` plugin: https://github.com/Cantera/cantera-website/blob/master/plugins/parse_docs.py You can more-or-less copy the implementation there to add a new identifier, in other words, add a `yaml_targets` dictionary to the `site` instance, and add the `yaml` folder to the loop at the bottom. That will get the targets processed into the cache. I think it only works for the current version docs, you might need to add a special case for YAML since its in the dev docs that we'll have to remove when we release the new version.

The section linking is done in the `ref.py` and `process_ref_targets.py` plugins. For the most part, the `ref.py` should just work if it finds a `:ref:` directive, so long as the location is stored in the cached `ref_targets` dictionary. So if you need to add new files/locations in the Science docs (the API docs are handled by the `parse_docs.py` plugin), just make sure that folder is processed in `process_ref_targets.py`.

I really need to document this...

Reply:
Well, I managed to get `parse_docs.py` to read through the dev docs, and I added an entry in `rest_class.py` so that the new role `ctidev:class` doesn't generate errors when used, but for some reason it produces links to the 2.4 docs instead of the dev. docs.

Reply:
Looking at `process_ref_targets.py`, I don't understand how it decides what folders to process. Does it process `:ref:` targets in the (stable) API docs? And how would this work if we wanted to have targets in the dev docs. Would this require another role to be introduced, e.g. `:devref:`?